### PR TITLE
Reorganize gulpfiles & gulp tasks

### PR DESCRIPTION
**Pull request by Swiip**
This is a big refactoring PR, I'll try to list my goals:

* enhance a little template interpretation to remove unnecessary blank lines #163 
* no more modifications on the `src` folder, injection target `.tmp` folder #202 #205 
* reorganize gulp files: add `scripts.js`, `styles.js` and `markups.js`, move injection and wiredep in `inject.js` #204 
* merge all injections in on task to avoid multiple access on `index.html`
* refactor of html preprocessor: handle only one preprocessor but simplify gulp code #213 
* use sub directories on `.tmp` folder to avoid any problems

### CODE REVIEW

File: app/src/write.js [hunk 1]
```diff
@@ -1,33 +1,54 @@
...
+        this.fs.copy(this.templatePath(src), this.destinationPath(dest));
+      }
+    } catch (error) {
+      console.error('Template processing error on file', src);
+      throw error;
+    }
+  }.bind(this);
+
+  // Copy dot files
+  _.forEach(files.dotFiles, function(src) {
```
Comment (marani) [inline app/src/write.js]: I think `dotfiles` is gone so this can be removed.

**Comment by marani** [comment]
It's neat I like the new gulp file templates.

File: app/src/write.js (hunk 1 above)
Comment (Swiip) [inline app/src/write.js]: You're right but I think I'll keep this waiting for a further refactoring PR...

**Comment by zckrs** [comment]
Huge PR but it's awesome
* New structure of gulp tasks is more readable and easier to change for a new generate project
* You fix lot of mistakes like multiple choices on htmlPreprocessor or the empty index.html
* Nice idea for lodash templating but you should add a comment on process function to explain why

Are you sure _.cloneDeep() create a new object and not use reference in mocha test ? Each test should have a independent prompt

**Comment by marani** [comment]
```
var characters = [
  { 'name': 'barney', 'age': 36 },
  { 'name': 'fred',   'age': 40 }
];

var deep = _.cloneDeep(characters);
deep[0] === characters[0];
// → false
```
as in lodash docs.

**Comment by Toilal** [comment]
:+1: Great thanks !

**Pull request merged**